tests/ui/useless_attribute.stderr

```diff
@@ -1,13 +1,13 @@
 error: useless lint attribute
-  --> $DIR/useless_attribute.rs:6:1
+  --> $DIR/useless_attribute.rs:7:1
    |
 LL | #[allow(dead_code)]
    | ^^^^^^^^^^^^^^^^^^^ help: if you just forgot a `!`, use: `#![allow(dead_code)]`
    |
    = note: `-D clippy::useless-attribute` implied by `-D warnings`
 
 error: useless lint attribute
-  --> $DIR/useless_attribute.rs:7:1
+  --> $DIR/useless_attribute.rs:8:1
```

Comment:
Does it also emit this suggestion if the attribute is not on top of the file? In that case, the suggestion would be wrong: [Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=06e0a1c60e27eed905bb31b8b52c124d)

Reply:
Yes it does :/ The following diff causes an incorrect suggestion:

```diff
diff --git a/tests/ui/useless_attribute.rs b/tests/ui/useless_attribute.rs
index c6fc5531..3b708dd9 100644
--- a/tests/ui/useless_attribute.rs
+++ b/tests/ui/useless_attribute.rs
@@ -1,5 +1,6 @@
 // run-rustfix
 // aux-build:proc_macro_derive.rs
+// aux-build:macro_rules.rs
 
 #![warn(clippy::useless_attribute)]
 #![warn(unreachable_pub)]
@@ -46,4 +47,7 @@ mod a {
     pub use self::b::C;
 }
 
+#[allow(dead_code)]
+extern crate macro_rules;
+
 fn main() {}
```

I wonder if we can detect if there have been _any_ items before the attribute. If yes, then we should not trigger the lint.

Reply:
I'm going to try and use the `items` field of [`hir::Crate`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/hir/struct.Crate.html) to check and see if the current item being checked is the first one. I'm not sure if that's the best way, though :thinking:

Reply:
Wait, `items` is a `BTreeMap`, so I'll have to check how/if that collection works wrt. sorting order :thinking: